README.md

```diff
@@ -21,3 +21,141 @@ From a command line, execute the application to get full usage help.
+
+```yaml
+svs:
+    ...
+    associated_images:
+        label:
+            action: replace
+            replace_with: blank_image
+    ...
+```
+
+This describes how `imagedephi` handles `label` images for Aperio files by default. Since label images frequently contain PHI, but are required by the Aperio (.svs) format, they are replaced with a black square of the same size.
+
+#### Image Rules
+
+Image rules take the following form:
+
+```yaml
+<image_key>:
+    action:
+```
+
+Where `image_key` identifies a particular associated image. For a catch-all rule, use the key `default`.
+
+Image rules can have the following actions:
+
+* `replace`: Replace an image with another. If specified, a value for `replace_with` must also be provided
+* `keep`: Does nothing. The associated image matching this key will be included in the output file
+* `delete`: The image will not be included in the output file
+
+For image rules, the only supported value of `replace_with` is `blank_image`.
+
+#### Metadata Rules
+
+Metadata rules take the following form:
+
+```yaml
+<metadata_key>:
+    action:
+```
+
+Where `metadata_key` identifies a piece of metadata. Possible values for this key depend on the type of metadata being redacted. For example, rules listed under
+
+```yaml
+tiff:
+    metadata:
+```
+have `metadata_keys` for particular tiff tags (e.g. `ImageDescription`, `ImageWidth`).
+
+Available actions for metadata rules are:
+
+* `delete`: the metadata will not appear in the output file
+* `keep`: the metadata will appear unchanged in the output file
+* `replace`: replace the metadata with a specified value. If this is the `action`, additional fields are required.
+* `check_type`: the metadata will appear in the output file if it is of the expected type. Requires additional fields
+
+##### `replace` rules
+Require the additional property `replace_with`. The value specified by the `replace_with` key will be used to override the metadata in the output image.
+
+##### `check_type` rules
+Use the additional properties:
+* `expected_type`: one of `integer`, `number`, `text`, `rational`
+* `expected_count` (optional): if the piece of metadata can contain multiple values, specify how many are expected using this property. Defaults to `1`
+
+### Supported Formats
+Currently, `imagedephi` supports redaction of pure `tiff` files as well as Aperio (.svs) files.
+
+Tiff rules have the following shape:
+
+```yaml
+tiff:
+    associated_images:
+        ...
+    metadata:
+        ...
+```
+
+The keys for the `metadata` rules are the names of tiff tags defined by the tiff standard.
+
+
+Aperio format rules have the following shape:
+
+```yaml
+svs:
+    associated_images:
+        ...
+    metadata:
+        ...
+    image_description:
+        ...
+```
+
+The keys for the `metadata` rules are the names of tiff tags defined by the tiff standard.
```

Comment:
In tifftools there are some variations of tag names to try to be inclusive to what other libraries do.  Are we only using the primary name or allowing any match?  I'd also comment on case sensitivity (or lack thereof) here.

Reply:
There's a [utility function ](https://github.com/DigitalSlideArchive/ImageDePHI/blob/46989ccde270bf725ad6fb6f8948fdc4e7ef7b65/imagedephi/utils/tiff.py#L4) that we use to get tiff tags. I think it satisfies allowing any match:

```python
>>> from imagedephi.utils.tiff import get_tiff_tag
>>> get_tiff_tag("GrayResponseUnit")
290
>>> get_tiff_tag("GreyResponseUnit")
290
```

Reply:
Great.  Let's say that `Names are case insensitive and common variations are accepted.`

Reply:
Done in cec9d21